data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconBlockHandler.java

```diff
@@ -13,57 +13,71 @@
 
 package tech.pegasys.artemis.beaconrestapi.beaconhandlers;
 
+import static javax.servlet.http.HttpServletResponse.SC_NOT_FOUND;
 import static tech.pegasys.artemis.datastructures.util.BeaconStateUtil.compute_start_slot_at_epoch;
 
-import com.google.common.collect.ImmutableMap;
 import com.google.common.primitives.UnsignedLong;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import org.apache.tuweni.bytes.Bytes32;
-import tech.pegasys.artemis.beaconrestapi.handlerinterfaces.BeaconRestApiHandler;
+import org.jetbrains.annotations.NotNull;
+import tech.pegasys.artemis.beaconrestapi.schema.BeaconBlockResponse;
 import tech.pegasys.artemis.datastructures.blocks.BeaconBlock;
 import tech.pegasys.artemis.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.artemis.provider.JsonProvider;
 import tech.pegasys.artemis.storage.ChainStorageClient;
 import tech.pegasys.artemis.storage.HistoricalChainData;
 
-public class BeaconBlockHandler implements BeaconRestApiHandler {
+public class BeaconBlockHandler implements Handler {
 
   private final ChainStorageClient client;
   private final HistoricalChainData historicalChainData;
+  public static final String ROUTE = "/beacon/block";
+  private final JsonProvider jsonProvider;
 
-  public BeaconBlockHandler(ChainStorageClient client, HistoricalChainData historicalChainData) {
+  public BeaconBlockHandler(
+      final ChainStorageClient client,
+      final HistoricalChainData historicalChainData,
+      final JsonProvider jsonProvider) {
     this.client = client;
     this.historicalChainData = historicalChainData;
+    this.jsonProvider = jsonProvider;
   }
 
   @Override
-  public String getPath() {
-    return "/beacon/block";
-  }
-
-  @Override
-  public Object handleRequest(RequestParams param) {
-    Map<String, List<String>> queryParamMap = param.getQueryParamMap();
+  public void handle(@NotNull Context ctx) throws Exception {
+    final Map<String, List<String>> queryParamMap = ctx.queryParamMap();
     if (queryParamMap.containsKey("root")) {
-      Bytes32 root = Bytes32.fromHexString(param.getQueryParam("root"));
-      return client.getStore() != null ? client.getStore().getBlock(root) : null;
+      final Bytes32 root = Bytes32.fromHexString(queryParamMap.get("root").get(0));
+      if (client.getStore() != null) {
+        ctx.result(
+            jsonProvider.objectToJSON(new BeaconBlockResponse(client.getStore().getBlock(root))));
```

Comment:
If the block doesn't exist this might get ugly, we might have to do the getBlock, check if its not null, then encode or 404...

Reply:
I'm thinking this could be where my current fail is.  Will investigate.